application-admintools-default/src/main/java/com/xwiki/admintools/internal/usage/InstanceUsageManager.java

```diff
@@ -148,106 +132,76 @@ public String renderTemplate()
     }
 
     /**
-     * Retrieves in descending order the documents that have more than a given number of comments.
+     * Get a {@link List} of {@link WikiSizeResult} with the options to sort it and apply filters on it.
      *
-     * @param maxComments maximum number of comments below which the document is ignored.
-     * @return a {@link List} with the documents that have more than the given number of comments.
-     * @throws QueryException if the query to retrieve the document fails.
+     * @param filters {@link Map} of filters to be applied on the gathered list.
+     * @param sortColumn target column to apply the sort on.
+     * @param order the order of the sort.
+     * @return a filtered and sorted {@link List} of {@link WikiSizeResult}.
      */
-    public List<String> getDocumentsOverGivenNumberOfComments(long maxComments) throws QueryException
+    public List<WikiSizeResult> getWikisSize(Map<String, String> filters, String sortColumn, String order)
     {
-        return this.queryManager.createQuery(
-                "select obj.name from BaseObject obj where obj.className='XWiki.XWikiComments' "
-                    + "group by obj.name having count(*) > :maxComments order by count(*) desc", Query.HQL)
-            .setWiki(wikiDescriptorManager.getCurrentWikiId()).bindValue("maxComments", maxComments).execute();
+        try {
+            Collection<WikiDescriptor> wikisDescriptors = getFilteredWikis(filters);
+            return usageDataProvider.getWikisSize(wikisDescriptors, filters, sortColumn, order);
```

Comment:
It feels a little strange that you are filtering the wikis, but then you send again these filters to the `getWikisSize` method, like you would filter twice. It would be best to maybe rename `getFilteredWikis` to not give the impression that you are doing extra filtering, but just taking the needed wikis. Maybe `getRequestedWikis` or simply `getWikis`?
Also, WDYT about doing this inside each provider method? The `getWikis` would be in the abstract class in this case.